Sources/LanguageServerProtocol/SupportTypes/CompletionItem.swift

```diff
@@ -11,7 +11,23 @@
 //===----------------------------------------------------------------------===//
 
 /// A single completion result.
-public struct CompletionItem: Codable, Hashable {
+public struct CompletionItem: TextDocumentRequest, RequestType, ResponseType, Codable, Hashable {
+  public typealias Response = CompletionItem
+    
+  public var textDocument: TextDocumentIdentifier {
+    if 
+      let data = self.data,
+      case let .dictionary(dict) = data,
+      case let .string(textDocURI) = dict["textDocURI"]
+    {
+      return TextDocumentIdentifier(DocumentURI(string: textDocURI))
+    } else {
+      fatalError("Missing textDocURI in CompletionItem.data: \(data ?? "(CompletionItem.data is nil)")")
+    }
+  }
+  
+  public static var method: String = "completionItem/resolve"
```

Comment:
I think the `method` property is always the first property in a request type. Could you move it up to match this style?